### DISCUSSION TITLE
EventSub via Websocket

### DISCUSSION THREAD

**Discussion by mrpainte** [TwitchIO/TwitchIO#350] [Feature requests]
Per the [post](https://discuss.dev.twitch.tv/t/eventsub-websockets-are-now-available-in-open-beta/41639) linked here is there any thought on creating a branch that implements this? i wouldn't mind assisting with this project portion but im unsure if that is warranted or not considering its still a "beta" option and will be for the next several months. I think with this addition it will help create the ability for non server owners the ability to create their own bots or create locally ran bots for streamers/moderators.

**Comment by IAmTomahawkx** [comment]
There's a long answer and a short answer to this. TL;DR we've moved the majority of our development time to some of our other projects which are in desperate need work, so development on this feature may be rather slow. 

Our endgoal is to make this transport the default, but for the time being it'll be on a branch once we start shifting back to TIO.

**Comment by IAmTomahawkx** [comment]
ongoing development can be tracked on the `feature/ws-eventsub` branch
Using this branch in any sort of non-testing environment is greatly discouraged

**Reply by mrpainte** [reply]
if i had greater understanding of this library i would love to help however i think it may just get in the way. I know on the pytwitchapi discussions someone asked if there can be documentation on how to use all the events, once this branch is considered to complete i would like to provide that documentation if you believe that it would be beneficial for this library. Thank you for all your work!!

**Reply by IAmTomahawkx** [reply]
@mrpainte "complete" is a strong word, but I don't believe we will be releasing this until eventsub websockets come out of beta. Feel free to contribute to the docs, I suck at them 🙃. In it's current state its mostly fine, I haven't received much feedback so I'm assuming it's working for people.

I'll say it now for future reference, a lot of the design in this feature is not fantastic because this library follows semver, so breaking changes can't be made without a bump in major version (and v3 is already in the works anyways). I've only realized now how terribad the eventsub ext was designed in terms of expandability, which is something i hope to fix in the complete redesign that is being worked on for v3
